storage_v2/v1adapter/ptrace2model_test.go

```diff
@@ -0,0 +1,149 @@
+
+				// Yield the test trace
+				yield([]ptrace.Traces{testTrace}, nil)
+			},
+			expectedErr: nil,
+		},
+		{
+			name: "seq with two chunks of ptrace.Traces",
+			expectedModelTraces: []*model.Trace{
+				{
+					Spans: []*model.Span{
+						{
+							TraceID:       model.NewTraceID(2, 3),
+							SpanID:        model.NewSpanID(1),
+							OperationName: "op-two-chunks-a",
+							Process:       model.NewProcess(ProcessNoServiceName, nil),
+							StartTime:     StartTime,
+						}, {
+							TraceID:       model.NewTraceID(2, 3),
+							SpanID:        model.NewSpanID(2),
+							OperationName: "op-two-chunks-b",
+							Process:       model.NewProcess(ProcessNoServiceName, nil),
+							StartTime:     StartTime,
+						},
+					},
+				},
+			},
+			seqTrace: func(yield func([]ptrace.Traces, error) bool) {
+				traceChunk1 := ptrace.NewTraces()
+				rSpans1 := traceChunk1.ResourceSpans().AppendEmpty()
+				sSpans1 := rSpans1.ScopeSpans().AppendEmpty()
+				spans1 := sSpans1.Spans()
+				modelTraceID := model.NewTraceID(2, 3)
+				span1 := spans1.AppendEmpty()
+				span1.SetTraceID(modelTraceID.ToOTELTraceID())
+				span1.SetName("op-two-chunks-a")
+				span1.SetSpanID(model.NewSpanID(1).ToOTELSpanID())
+
+				traceChunk2 := ptrace.NewTraces()
+				rSpans2 := traceChunk2.ResourceSpans().AppendEmpty()
+				sSpans2 := rSpans2.ScopeSpans().AppendEmpty()
+				spans2 := sSpans2.Spans()
+				span2 := spans2.AppendEmpty()
+				span2.SetTraceID(modelTraceID.ToOTELTraceID())
+				span2.SetName("op-two-chunks-b")
+				span2.SetSpanID(model.NewSpanID(2).ToOTELSpanID())
+				// Yield the test trace
+				yield([]ptrace.Traces{traceChunk1, traceChunk2}, nil)
+			},
+			expectedErr: nil,
+		},
+		{
+			// a case that occurs when no trace is contained in the iterator
+			name:                "empty seq for ptrace.Traces",
+			expectedModelTraces: nil,
+			seqTrace: func(yield func([]ptrace.Traces, error) bool) {
+			},
+			expectedErr: spanstore.ErrTraceNotFound,
+		},
+		{
+			name: "seq of one ptrace and one error",
+			expectedModelTraces: nil,
+			seqTrace: func(yield func([]ptrace.Traces, error) bool) {
+				testTrace := ptrace.NewTraces()
+				rSpans := testTrace.ResourceSpans().AppendEmpty()
+				sSpans := rSpans.ScopeSpans().AppendEmpty()
+				spans := sSpans.Spans()
+
+				modelTraceID := model.NewTraceID(2, 3)
+				span1 := spans.AppendEmpty()
+				span1.SetTraceID(modelTraceID.ToOTELTraceID())
+				span1.SetName("op-error-a")
+				span1.SetSpanID(model.NewSpanID(1).ToOTELSpanID())
+
+				// Yield the test trace
+				if !yield([]ptrace.Traces{testTrace}, nil) {
+					return
+				}
+				yield(nil, errors.New("unexpected-op-err"))
+			},
+			expectedErr: errors.New("unexpected-op-err"),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			actualTraces, err := PTracesSeq2ToModel(tc.seqTrace)
+			require.Equal(t, tc.expectedErr, err)
+			CompareSliceOfTraces(t, tc.expectedModelTraces, actualTraces)
```

Comment:
you don't need to compare full traces, it's not important for the tests you're doing. It's sufficient to compare some other basic collections, like a list of trace/span IDs or span names. This way you don't need to depend on a complex helper function from another package (and create a cross-package dependencies)